### PR TITLE
AO3-6010 Add role for Legal

### PR DESCRIPTION
**Pull request by brianjaustin**
# Pull Request Checklist

## Issue

https://otwarchive.atlassian.net/browse/AO3-6010

## Purpose

Add a role for our Legal folks, along with some permissions for that role

## Credit

Brian Austin (they/he)

### CODE REVIEW

File: features/admins/admin_works.feature
Comment (Bilka2) [inline features/admins/admin_works.feature:125]: Is this step needed?

File: features/admins/admin_works.feature
Comment (Bilka2) [inline features/admins/admin_works.feature:139]: I think logging in as bad_user would be better, to distinguish from hiding the bookmark.

File: features/admins/admin_works.feature
Comment (Bilka2) [inline features/admins/admin_works.feature:235]: Is this step needed?

File: spec/controllers/comments_controller_spec.rb
Comment (Bilka2) [inline spec/controllers/comments_controller_spec.rb:2390]: In line 2456 it looks like legal should also be added.

File: spec/controllers/comments_controller_spec.rb
Comment (Bilka2) [inline spec/controllers/comments_controller_spec.rb:2560]: In line 2626 it looks like legal should also be added.

File: spec/controllers/comments_controller_spec.rb
Comment (Bilka2) [inline spec/controllers/comments_controller_spec.rb:2732]: In line 2798 it looks like legal should also be added.

File: spec/controllers/comments_controller_spec.rb
Comment (Bilka2) [inline spec/controllers/comments_controller_spec.rb:2902]: In line 2968 it looks like legal should also be added.

File: spec/controllers/comments_controller_spec.rb
Comment (Bilka2) [inline spec/controllers/comments_controller_spec.rb:3286]: This array and the one in line 3299 could be consolidated into a variable.

File: spec/controllers/comments_controller_spec.rb
Comment (Bilka2) [inline spec/controllers/comments_controller_spec.rb:3404]: This array and the one in line 3417 could be consolidated into a variable.

File: spec/controllers/comments_controller_spec.rb
Comment (Bilka2) [inline spec/controllers/comments_controller_spec.rb:3417]: In lines 3851 it looks like legal should also be added.

File: spec/controllers/admin/admin_users_controller_spec.rb
Comment (Bilka2) [inline spec/controllers/admin/admin_users_controller_spec.rb:36]: This text looks like it needs an update.

File: spec/controllers/admin/admin_users_controller_spec.rb
Comment (Bilka2) [inline spec/controllers/admin/admin_users_controller_spec.rb:75]: There is also this option:
```suggestion
      let(:admin) { create(:policy_and_abuse_admin) }
```

**Review by Bilka2 — CHANGES_REQUESTED** [review]
Two questions regarding mismatches between Jira and what was implemented and a few notes for the tests.

`admin_users_controller_spec.rb` is missing a few tests:
* test that update doesn't allow legal to change email or roles
* test that update_next_of_kin, update_status, confirm_delete_user_creations, destroy_user_creations doesn't allow the unauthorized admin roles to do changes/access the actions
* activate and troubleshoot aren't testing any roles besides support

File: app/policies/comment_policy.rb
Comment (brianjaustin) [reply app/policies/comment_policy.rb:8]: This is for approve/reject, so if not spam then this would need to be broken apart further. To me, though, those imply spam as well. Additionally, both support and PAC have this permission, so I don't see a reason to block Legal from marking things as spam/not spam

File: app/policies/user_policy.rb
Comment (brianjaustin) [reply app/policies/user_policy.rb:7]: I think Legal having access to those is fine, but I'll defer to @sarken on that one

File: app/policies/comment_policy.rb
Comment (Bilka2) [reply app/policies/comment_policy.rb:8]: approve/reject seems to be for spam too. Are you maybe thinking of `can_review_comment?` for "approve moderated admin post comments"? That one is independent of role and always allowed.

But I agree that it sound off to allow legal to delete admin post comments, but not spam/ham admin post comments. So this also sounds like @sarken should take a look.

File: app/policies/user_policy.rb
Comment (sarken) [reply app/policies/user_policy.rb:7]: Yeah, it's fine for Legal to have access to those, but we should update the issue to mention it.

File: app/policies/comment_policy.rb
Comment (sarken) [reply app/policies/comment_policy.rb:8]: Bilka is right that the issue is referring to moderated comments, meaning this: https://github.com/otwcode/otwarchive/blob/cd20f8e66c426944136f4673c35c70c5a6333548/app/policies/comment_policy.rb#L51-L57 
And it's not something Legal explicitly asked for; it just comes as a little side benefit of being an admin (for now -- we'll eventually limit which admins can do that, but BAT/Comms/NPM were still figuring out workflow and purview when we implemented moderated news post comments).

While there's no _harm_ in giving Legal the ability to spam comments, it's not something they're going to use, so let's leave them out of `SPAM_ADMIN_POST_COMMENT_ROLES`.

File: features/admins/admin_works.feature
Comment (brianjaustin) [reply features/admins/admin_works.feature]: Appears not, :wastebasket:

File: features/admins/admin_works.feature
Comment (brianjaustin) [reply features/admins/admin_works.feature]: Appears not, 🗑️

File: app/policies/user_policy.rb
Comment (brianjaustin) [reply app/policies/user_policy.rb:7]: :+1: Added a note to the Jira bullet point about user admin pages

**Review by Bilka2 — APPROVED** [review]
Thank you!

**Comment by Bilka2** [comment]
Wait, sorry, I missed my own comment regarding tests in `admin_users_controller_spec`. I think the "test that update doesn't allow legal to change email or roles" is needed, since no other admin role has this behaviour currently. The other two points aren't as important, since legal is treading any new ground there.

**Review by Bilka2 — APPROVED** [review]
Thank you!

**Pull request merged**
